pkg/controller/common.go

```diff
@@ -686,6 +688,21 @@ func getRankedNodesForReplicaAttachments(ctx context.Context, azr azReconciler,
 		if pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil {
 			continue
 		}
+
+		// Find the intersection of the zones for all the volumes
+		if i == 0 {
+			compatibleZonesMap = getSupportedZones(pv.Spec.NodeAffinity.Required.NodeSelectorTerms)
+		} else {
+			commonZones := map[string]bool{}
```

Comment:
Since the mapped value isn't used or meaningful, you can use an empty struct here, i.e:

```go
commonZones := make(map[string]struct{}).
```

You add an entry like so:

```go
commonZones[key] = struct{}{}
```

Reply:
Done

Reply:
Instead of making a `map[string]struct{}{}`, maybe we can use set type defined in `common.go`?https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/4c6fe4fb942ae2ecc70df15c0073b0a87dcce02a/pkg/controller/common.go#L194:L196

Reply:
changed to use set